pkg/constants/http.go

```diff
@@ -207,3 +225,13 @@ func URLForGetMostRecentNodeConfig(nodeID string, version *types.Version) string
 	return ProvenanceEndpoint + path.Join("node", nodeID) +
 		fmt.Sprintf("?blocknumber=%d&transactionnumber=%d", version.BlockNum, version.TxNum)
 }
+
+// SafeURLSegmentNZ checks that the string `s` is safe to use as a URL segment-nz.
+// For example: `http://example.com:8080/tx/my-id`, for s="my-id".
+// See: `https://www.ietf.org/rfc/rfc3986.txt`.
+func SafeURLSegmentNZ(s string) error {
+	if !validURLSegmentNZ.MatchString(s) {
+		return errors.Errorf("un-safe for a URL segment: %q", s)
```

Comment:
%q prints unprintable chars as the binary 0x10 code, so that if there is an error and `s` is not printable, the output actually gives info about what is wrong, instead of emitting the strange char that says "unprintable" (black diamond with ?, or black square with 1234 in it, depending on the platform).

Reply:
Why `5q` and not `%s`?